airflow/api_fastapi/views/public/dags.py

```diff
@@ -71,16 +72,13 @@ async def get_dags(
 
     dags = session.scalars(dags_query).all()
 
-    try:
-        return DAGCollectionResponse(
-            dags=[DAGResponse.model_validate(dag, from_attributes=True) for dag in dags],
-            total_entries=total_entries,
-        )
-    except ValueError as e:
-        raise HTTPException(400, f"DAGCollectionSchema error: {str(e)}")
+    return DAGCollectionResponse(
+        dags=[DAGResponse.model_validate(dag, from_attributes=True) for dag in dags],
+        total_entries=total_entries,
+    )
 
 
-@dags_router.patch("/dags/{dag_id}")
+@dags_router.patch("/dags/{dag_id}", responses=create_openapi_http_exception_doc([400, 401, 403, 404]))
```

Comment:
This is how we tell FastAPI the possible error code returned by the endpoint

Reply:
Interesting that this is not natively supported. Was it that our previous approach was not according to design of FastAPI or is this missin gin the copcepts?

Reply:
I mean the `responses` field is supported and we can document them, but without this utility function that would be a lot of manual and duplicated typing.

FastAPI does not automatically detect and document HTTPExceptions, so yes definitely missing from FastAPI. This discussions https://github.com/fastapi/fastapi/discussions/9124#discussioncomment-8571860 explains it well I think.